service/dap/server.go

```diff
@@ -469,6 +469,123 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		return
 	}
 
+	if mode == "record" {
+		s.config.Debugger.Backend = "rr"
+		s.send(&dap.CapabilitiesEvent{ Event: *newEvent("capabilities"), Body: dap.CapabilitiesEventBody{Capabilities: dap.Capabilities{ SupportsStepBack: true }}})
+
+		// Validate rr availability
+		if err := gdbserial.CheckRRAvailable(); err != nil {
+			s.sendErrorResponseWithURL(request.Request, FailedToInitialize, "Unable to find  rr in record configuration", err.Error(), "https://github.com/rr-debugger/rr/wiki/Building-And-Installing", "rr installation page")
+			return
+		}
+
+		// Record only works on rr trace directories
+		traceDirectory, ok := request.Arguments["traceDirectory"].(string)
```

Comment:
You can't tell which trace directory to use to `gdbserial.Record` (actually, I don't think you can tell `rr` which trace directory to use, but maybe I'm wrong). This is making the user specify a directory that isn't ever used.